src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs

```diff
@@ -759,14 +759,18 @@ private List<PortViewModel> CreateProxyOutPorts(IEnumerable<PortModel> groupPort
                 .ToList();
 
             var newPortViewModels = new List<PortViewModel>();
-            for (int i = 0; i < groupPortModels.Count(); i++)
+            int index = 0;
+            foreach (var group in groupPortModels)
             {
-                var model = groupPortModels.ElementAt(i);
-                newPortViewModels.Add(originalPortViewModels[i].CreateProxyPortViewModel(model));
-
-                // calculate new position for the proxy outports
-                model.Center = CalculatePortPosition(model, i);
-                model.Owner.ReportPosition();
+                var originalPort = originalPortViewModels.FirstOrDefault(x => x.PortModel.GUID == group.GUID);
```

Comment:
I am just curious, how does this make a difference. Weren't were iterating through all the output-ports in the first place? or is this just refactoring?

Reply:
Hi, thanks for the observation. Previously we were interacting through all the ports without controlling its owner. Some ports were wrongly set to another group or node, causing the bug. By searching its GUID we make sure the right node/group is referenced.